packages/compass-aggregations/src/modules/input-documents.ts

```diff
@@ -122,14 +122,23 @@ export const refreshInputDocuments = (): PipelineBuilderThunkAction<
         | undefined,
     };
 
+    // maxTimeMS defaults to null here because the aggregation options field's
+    // maxTimeMS defaults to empty and the preference defaults to undefined.
+    // We need a timeout on count because for timeseries estimatedCount() seems
+    // to just do a colscan and we need that timeout to be low compared to the
+    // aggregation one anyway. For aggregations it is less critical because we
+    // $limit to the first few records.
+    const countOptions = { ...options, maxTimeMS: 5000 };
```

Comment:
I thought about removing this again now that I also changed the method in DataService to fill it in if it isn't set, but I think the comment is useful documentation and it might still be a good idea to decouple this timeout from the hardcoded default in DataService. Yes right now it is the same, but now it is easy for someone to change the value we use in aggregations.

Reply:
Oh and it also means that we print a value for maxTimeMS in the logs rather than null so no one will get a heart attack.

Reply:
https://jira.mongodb.org/browse/COMPASS-7798